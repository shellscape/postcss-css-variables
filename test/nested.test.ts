import { fixture, process } from './helpers';

const tests = {
  'should not mangle outer function parentheses': 'nested-inside-other-func',
  'should not mangle outer function parentheses - calc': 'nested-inside-calc-func',
  'should not mangle outer function parentheses - calc with fallback':
    'nested-inside-calc-func-with-fallback',
  'should not mangle outer function parentheses - calc with fallback var()':
    'nested-inside-calc-func-with-fallback-var',
  'should not mangle outer function parentheses - with fallback':
    'nested-inside-other-func-with-fallback',
  'should work with nested at-rules containing properties': 'nested-at-rules-containing-properties'
};

describe('nested', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`nested/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });
});
