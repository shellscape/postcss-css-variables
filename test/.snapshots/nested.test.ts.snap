// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nested > should not mangle outer function parentheses - calc 1`] = `".box-foo{width:calc(1000% - 150px)}.box-foo{width:undefined}.box-foo{width:calc(150px - 50px)}.box-foo{width:calc(calc(100px / 2) / 2)}"`;

exports[`nested > should not mangle outer function parentheses - calc with fallback 1`] = `".box-foo{width:calc(58.3333333333% - 150px)}.box-foo{width:calc(58.3333333333% - 100px)}"`;

exports[`nested > should not mangle outer function parentheses - calc with fallback var() 1`] = `".box-foo{width:calc(58.3333333333% - 150px)}.box-foo{width:calc(58.3333333333% - 100px)}.box-foo{width:calc(58.3333333333% - 50px)}"`;

exports[`nested > should not mangle outer function parentheses - with fallback 1`] = `".box-foo{background-color:color(white)}.box-foo{background-color:color(red)}"`;

exports[`nested > should not mangle outer function parentheses 1`] = `".box-foo{background-color:color(red)}.box-foo{background-color:rgba(255,0,0,0.3)}.box-foo{background-color:undefined}"`;

exports[`nested > should work with nested at-rules containing properties 1`] = `"@page{@footnote{background-color:red}}"`;
